### PR TITLE
rust volume: verify the .dat ends at the last indexed needle

### PR DESCRIPTION
**Pull request by chrislusf**
The Go loader quarantines a volume whose .dat extends past the last indexed needle - the leftover of a torn shutdown - but the Rust check only verified header fields of the trailing index entries and never compared file sizes, so a torn tail loaded clean and writable. That is worse than a missed quarantine: appends land at the raw file end, and past a misaligned tail the next needle sits at an offset the 8-byte-unit .idx encoding rounds down, pointing the index a few bytes before the needle - every read of it then fails. Even with an aligned tail, a mixed replica set diverges: the Go replica goes read-only while the Rust one keeps writing.

Replace the last-10-entries walk with the current Go shape: find the entry physically last in the .dat (append-ordered fast path, max-offset scan for key-sorted rebuilds), verify that needle - tombstones with their on-disk Size=0 - and require the file to end exactly at it, marking the volume read-only otherwise.

Tests: torn tail after a live needle, torn tail after a tombstone, key-sorted .idx does not falsely flip a healthy volume, truncated .dat quarantines. Full suite passes.

https://claude.ai/code/session_01XgGXMLknzaNgQzHMyo2Vhb

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved storage volume integrity checks to verify the physical end of data files, ensuring detection of trailing garbage and truncated mid-record tails.
  * Correctly validates the last live record versus deletion tombstones, including large-file and appended-timestamp edge cases.
  * Reduces incorrect “read-only quarantine” outcomes for otherwise valid volumes with certain index layouts.
* **Tests**
  * Expanded integrity regression coverage with new end-of-file corruption scenarios and multi-needle reload validation.

### CODE REVIEW

File: seaweed-volume/src/storage/volume.rs [hunk 1]
```diff
@@ -1866,104 +1866,213 @@ impl Volume {
...
+            if !offset.is_zero() {
+                let ao = offset.to_actual_offset();
+                if ao > max_actual_offset {
+                    max_actual_offset = ao;
+                    max_entry_pos = entry_pos;
                 }
-                Err(e) => return Err(e),
             }
+            entry_pos += NEEDLE_MAP_ENTRY_SIZE as i64;
+        }
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply seaweed-volume/src/storage/volume.rs:1947]: Took the BufReader — the scan is now buffered like Go's WalkIndexFile.

Not skipping tombstone entries though: a trailing deletion is a real on-disk record (Size=0 needle) and can be the one physically last in the .dat — filtering it out would compare a mid-file live needle against the full file size and falsely quarantine any volume whose last write was a delete. Go's findDatTailEntryOffset keeps them for the same reason, and test_check_volume_data_integrity_with_deletion_tombstone pins the behavior.

**Comment by chrislusf** [comment]
The BufReader nitpick is addressed in 33c9676eb.
